clusterman/draining/queue.py

```diff
@@ -388,34 +391,28 @@ def process_drain_queue(
                         self.submit_host_for_termination(host_to_process, delay=0)
                     else:  # case 0c
                         logger.info(f"Sending host to drain: {host_to_process.instance_id}")
-                        # message shouldn't add to cache since it will be received again.
-                        should_add_to_cache = False
-                        self.submit_host_for_draining(host_to_process_fresh)
+                        self.submit_host_for_draining(host_to_process_fresh, host_to_process.attempt + 1)
                 elif spent_time.total_seconds() > draining_time_threshold_seconds:
                     if force_terminate:  # case 1
                         logger.info(f"Draining expired for: {host_to_process.instance_id}")
                         self.submit_host_for_termination(host_to_process, delay=0)
                     else:  # case 2
                         k8s_uncordon(kube_operator_client, host_to_process.agent_id)
-                        should_add_to_cache = False
+                        #  removing instance_id from cache to avoid unnecessary blocking by cache
+                        self.draining_host_ttl_cache.pop(host_to_process.instance_id, None)
                 else:  # case 3
-                    should_add_to_cache = self._drain_k8s_host(kube_operator_client, host_to_process, disable_eviction)
-                    should_resend_to_queue = not should_add_to_cache
-
-                if should_resend_to_queue:
-                    logger.info(
-                        f"Delaying re-draining {host_to_process.instance_id} for {redraining_delay_seconds} seconds"
-                    )
-                    self.submit_host_for_draining(host_to_process, redraining_delay_seconds)
+                    if not self._drain_k8s_host(kube_operator_client, host_to_process, disable_eviction):
```

Comment:
This can become an `elif` and save 1 block of indentation.

Reply:
@piax93 done.